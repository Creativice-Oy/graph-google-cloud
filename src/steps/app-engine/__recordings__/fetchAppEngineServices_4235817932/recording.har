{
  "log": {
    "_recordingName": "fetchAppEngineServices",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 565,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 565,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:57:50 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:57:50.653Z",
        "time": 170,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 170
        }
      },
      {
        "_id": "a8077ca10a73994713b7a0d586480166",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "appengine.googleapis.com"
            }
          ],
          "headersSize": 538,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://appengine.googleapis.com/v1/apps/j1-gc-integration-dev-v2"
        },
        "response": {
          "bodySize": 637,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 637,
            "text": "{\"name\":\"apps/j1-gc-integration-dev-v2\",\"id\":\"j1-gc-integration-dev-v2\",\"authDomain\":\"gmail.com\",\"locationId\":\"us-central\",\"codeBucket\":\"staging.j1-gc-integration-dev-v2.appspot.com\",\"servingStatus\":\"SERVING\",\"defaultHostname\":\"j1-gc-integration-dev-v2.uc.r.appspot.com\",\"defaultBucket\":\"j1-gc-integration-dev-v2.appspot.com\",\"gcrDomain\":\"us.gcr.io\",\"databaseType\":\"CLOUD_DATASTORE_COMPATIBILITY\",\"featureSettings\":{\"splitHealthChecks\":true,\"useContainerOptimizedOs\":true}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:57:51 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:57:50.835Z",
        "time": 455,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 455
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 552,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 552,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:58:50 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:58:49.917Z",
        "time": 169,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 169
        }
      },
      {
        "_id": "83742efc2d14453e4066991a3f93e347",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "appengine.googleapis.com"
            }
          ],
          "headersSize": 547,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://appengine.googleapis.com/v1/apps/j1-gc-integration-dev-v2/services"
        },
        "response": {
          "bodySize": 311,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 311,
            "text": "{\"services\":[{\"name\":\"apps/j1-gc-integration-dev-v2/services/default\",\"id\":\"default\",\"split\":{\"allocations\":{\"20210309t141022\":1}}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Thu, 11 Mar 2021 14:58:50 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-03-11T14:58:50.091Z",
        "time": 461,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 461
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
