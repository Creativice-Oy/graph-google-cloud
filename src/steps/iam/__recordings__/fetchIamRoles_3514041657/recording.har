{
  "log": {
    "_recordingName": "fetchIamRoles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 735,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "735"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 541,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 541,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 05 Aug 2020 15:12:48 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-27=\":443\"; ma=2592000,h3-T050=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 536,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2020-08-05T15:12:48.687Z",
        "time": 97,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 97
        }
      },
      {
        "_id": "7f5acee88c214a4d344913bc19611940",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.0 gl-node/13.10.1 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.0 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "iam.googleapis.com"
            }
          ],
          "headersSize": 531,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "parent",
              "value": "projects/j1-gc-integration-dev"
            },
            {
              "name": "view",
              "value": "FULL"
            }
          ],
          "url": "https://iam.googleapis.com/v1/roles?parent=projects%2Fj1-gc-integration-dev&view=FULL"
        },
        "response": {
          "bodySize": 467,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 467,
            "text": "{\"roles\":[{\"name\":\"projects/j1-gc-integration-dev/roles/545240943112customrole\",\"title\":\"GA custom role\",\"description\":\"My description\",\"includedPermissions\":[\"cloudfunctions.functions.get\",\"cloudfunctions.functions.getIamPolicy\"],\"stage\":\"GA\",\"etag\":\"BwWsEa2N+wo=\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 05 Aug 2020 15:12:49 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-27=\":443\"; ma=2592000,h3-T050=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 513,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2020-08-05T15:12:48.802Z",
        "time": 329,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 329
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
