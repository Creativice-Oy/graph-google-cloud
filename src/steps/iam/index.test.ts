import {
  Recording,
  createMockStepExecutionContext,
} from '@jupiterone/integration-sdk-testing';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchIamRoles, fetchIamServiceAccounts } from '.';
import { integrationConfig } from '../../../test/config';

describe('#fetchIamRoles', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchIamRoles',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchIamRoles(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(context.jobState.collectedEntities).toMatchGraphObjectSchema({
      _class: ['AccessRole'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_iam_role' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          description: { type: 'string' },
          stage: { type: 'string' },
          custom: { const: true },
          deleted: { type: 'boolean' },
          permissions: {
            type: 'array',
            items: { type: 'string' },
          },
          etag: { type: 'string' },
        },
      },
    });
  });
});

describe('#fetchIamServiceAccounts', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchIamServiceAccounts',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchIamServiceAccounts(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === 'google_iam_service_account',
      ),
    ).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_iam_service_account' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          enabled: { type: 'boolean' },
          projectId: { type: 'string' },
          id: { type: 'string' },
          oauth2ClientId: { type: 'string' },
          etag: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === 'google_iam_service_account_key',
      ),
    ).toMatchGraphObjectSchema({
      _class: ['AccessKey'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_iam_service_account_key' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          origin: { type: 'string' },
          type: { type: 'string' },
          algorithm: { type: 'string' },
        },
      },
    });

    expect(context.jobState.collectedRelationships).toEqual(
      context.jobState.collectedRelationships.map((r) =>
        expect.objectContaining({
          _class: 'HAS',
          webLink: expect.any(String),
        }),
      ),
    );
  });
});
