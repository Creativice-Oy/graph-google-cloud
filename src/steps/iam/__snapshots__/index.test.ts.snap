// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchIamRoles should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "google_cloud_iam_role_projects/j1-gc-integration-dev/roles/545240943112customrole",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "My description",
            "etag": "BwWsEa2N+wo=",
            "includedPermissions": Array [
              "cloudfunctions.functions.get",
              "cloudfunctions.functions.getIamPolicy",
            ],
            "name": "projects/j1-gc-integration-dev/roles/545240943112customrole",
            "stage": "GA",
            "title": "GA custom role",
          },
        },
      ],
      "_type": "google_cloud_iam_role",
      "createdOn": undefined,
      "custom": true,
      "deleted": false,
      "description": "My description",
      "displayName": "GA custom role",
      "etag": "BwWsEa2N+wo=",
      "name": "projects/j1-gc-integration-dev/roles/545240943112customrole",
      "permissions": Array [
        "cloudfunctions.functions.get",
        "cloudfunctions.functions.getIamPolicy",
      ],
      "stage": "GA",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_cloud_iam_role",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchIamServiceAccounts should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "google_cloud_iam_service_account_projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "J1 Google Cloud integration execution",
            "email": "j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
            "etag": "MDEwMjE5MjA=",
            "name": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
            "oauth2ClientId": "116742477893551302665",
            "projectId": "j1-gc-integration-dev",
            "uniqueId": "116742477893551302665",
          },
        },
      ],
      "_type": "google_cloud_iam_service_account",
      "createdOn": undefined,
      "description": "J1 Google Cloud integration execution",
      "displayName": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
      "email": "j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
      "enabled": true,
      "etag": "MDEwMjE5MjA=",
      "id": "116742477893551302665",
      "name": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
      "oauth2ClientId": "116742477893551302665",
      "projectId": "j1-gc-integration-dev",
      "username": "j1-gc-integration-dev-sa@j1-gc-integration-dev.iam.gserviceaccount.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "google_cloud_iam_service_account_projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev@appspot.gserviceaccount.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "App Engine default service account",
            "email": "j1-gc-integration-dev@appspot.gserviceaccount.com",
            "etag": "MDEwMjE5MjA=",
            "name": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev@appspot.gserviceaccount.com",
            "oauth2ClientId": "117962362686380350362",
            "projectId": "j1-gc-integration-dev",
            "uniqueId": "117962362686380350362",
          },
        },
      ],
      "_type": "google_cloud_iam_service_account",
      "createdOn": undefined,
      "description": undefined,
      "displayName": "App Engine default service account",
      "email": "j1-gc-integration-dev@appspot.gserviceaccount.com",
      "enabled": true,
      "etag": "MDEwMjE5MjA=",
      "id": "117962362686380350362",
      "name": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev@appspot.gserviceaccount.com",
      "oauth2ClientId": "117962362686380350362",
      "projectId": "j1-gc-integration-dev",
      "username": "j1-gc-integration-dev@appspot.gserviceaccount.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "google_cloud_iam_service_account_projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "J1 Google Cloud integration Terraform setup",
            "email": "j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
            "etag": "MDEwMjE5MjA=",
            "name": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
            "oauth2ClientId": "116674556692942636222",
            "projectId": "j1-gc-integration-dev",
            "uniqueId": "116674556692942636222",
          },
        },
      ],
      "_type": "google_cloud_iam_service_account",
      "createdOn": undefined,
      "description": "J1 Google Cloud integration Terraform setup",
      "displayName": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
      "email": "j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
      "enabled": true,
      "etag": "MDEwMjE5MjA=",
      "id": "116674556692942636222",
      "name": "projects/j1-gc-integration-dev/serviceAccounts/j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
      "oauth2ClientId": "116674556692942636222",
      "projectId": "j1-gc-integration-dev",
      "username": "j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_cloud_iam_service_account",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 0,
}
`;
